File: backend/Models/Boleto.cs
using System.ComponentModel.DataAnnotations;

namespace CrmArrighi.Models
{
    public class Boleto
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "O contrato é obrigatório")]
        public int ContratoId { get; set; }
        public Contrato Contrato { get; set; } = null!;

        [Required(ErrorMessage = "O NSU Code é obrigatório")]
        [StringLength(20, ErrorMessage = "NSU Code deve ter no máximo 20 caracteres")]
        public string NsuCode { get; set; } = string.Empty;

        [Required(ErrorMessage = "A data NSU é obrigatória")]
        public DateTime NsuDate { get; set; }

        [Required(ErrorMessage = "O código do convênio é obrigatório")]
        [StringLength(9, ErrorMessage = "Código do convênio deve ter no máximo 9 caracteres")]
        public string CovenantCode { get; set; } = string.Empty;

        [Required(ErrorMessage = "O nosso número é obrigatório")]
        [StringLength(13, ErrorMessage = "Nosso número deve ter no máximo 13 caracteres")]
        public string BankNumber { get; set; } = string.Empty;

        [StringLength(15, ErrorMessage = "Seu número deve ter no máximo 15 caracteres")]
        public string? ClientNumber { get; set; }

        [Required(ErrorMessage = "A data de vencimento é obrigatória")]
        public DateTime DueDate { get; set; }

        [Required(ErrorMessage = "A data de emissão é obrigatória")]
        public DateTime IssueDate { get; set; }

        [Required(ErrorMessage = "O valor nominal é obrigatório")]
        [Range(0.01, 999999999.99, ErrorMessage = "Valor nominal deve ser maior que zero")]
        public decimal NominalValue { get; set; }

        [Required(ErrorMessage = "O tipo de documento é obrigatório")]
        [StringLength(50, ErrorMessage = "Tipo de documento deve ter no máximo 50 caracteres")]
        public string DocumentKind { get; set; } = string.Empty;

        // Dados do Pagador
        [Required(ErrorMessage = "O nome do pagador é obrigatório")]
        [StringLength(40, ErrorMessage = "Nome do pagador deve ter no máximo 40 caracteres")]
        public string PayerName { get; set; } = string.Empty;

        [Required(ErrorMessage = "O tipo de documento do pagador é obrigatório")]
        [StringLength(4, ErrorMessage = "Tipo de documento deve ser CPF ou CNPJ")]
        public string PayerDocumentType { get; set; } = string.Empty;

        [Required(ErrorMessage = "O número do documento do pagador é obrigatório")]
        [StringLength(15, ErrorMessage = "Número do documento deve ter no máximo 15 caracteres")]
        public string PayerDocumentNumber { get; set; } = string.Empty;

        [Required(ErrorMessage = "O endereço do pagador é obrigatório")]
        [StringLength(40, ErrorMessage = "Endereço deve ter no máximo 40 caracteres")]
        public string PayerAddress { get; set; } = string.Empty;

        [Required(ErrorMessage = "O bairro do pagador é obrigatório")]
        [StringLength(30, ErrorMessage = "Bairro deve ter no máximo 30 caracteres")]
        public string PayerNeighborhood { get; set; } = string.Empty;

        [Required(ErrorMessage = "A cidade do pagador é obrigatória")]
        [StringLength(20, ErrorMessage = "Cidade deve ter no máximo 20 caracteres")]
        public string PayerCity { get; set; } = string.Empty;

        [Required(ErrorMessage = "O estado do pagador é obrigatório")]
        [StringLength(2, ErrorMessage = "Estado deve ter 2 caracteres")]
        public string PayerState { get; set; } = string.Empty;

        [Required(ErrorMessage = "O CEP do pagador é obrigatório")]
        [StringLength(9, ErrorMessage = "CEP deve ter formato 00000-000")]
        public string PayerZipCode { get; set; } = string.Empty;

        // Campos opcionais para desconto, multa e juros
        public decimal? FinePercentage { get; set; }
        public int? FineQuantityDays { get; set; }
        public decimal? InterestPercentage { get; set; }
        public decimal? DeductionValue { get; set; }
        public int? WriteOffQuantityDays { get; set; }

        // Campos de protesto
        /// <summary>
        /// Tipo de protesto: DIAS_CORRIDOS, DIAS_UTEIS, SEM_PROTESTO, NAO_PROTESTAR
        /// </summary>
        [StringLength(20, ErrorMessage = "Tipo de protesto deve ter no máximo 20 caracteres")]
        public string? ProtestType { get; set; }

        /// <summary>
        /// Quantidade de dias para protesto (após vencimento)
        /// </summary>
        [Range(1, 99, ErrorMessage = "Dias para protesto deve estar entre 1 e 99")]
        public int? ProtestQuantityDays { get; set; }

        // Campos de resposta da API Santander
        [StringLength(100, ErrorMessage = "Código de barras deve ter no máximo 100 caracteres")]
        public string? BarCode { get; set; }

        [StringLength(100, ErrorMessage = "Linha digitável deve ter no máximo 100 caracteres")]
        public string? DigitableLine { get; set; }

        public DateTime? EntryDate { get; set; }

        [StringLength(500, ErrorMessage = "QR Code PIX deve ter no máximo 500 caracteres")]
        public string? QrCodePix { get; set; }

        [StringLength(500, ErrorMessage = "URL QR Code deve ter no máximo 500 caracteres")]
        public string? QrCodeUrl { get; set; }

        // Status do boleto
        [Required(ErrorMessage = "O status é obrigatório")]
        [StringLength(20, ErrorMessage = "Status deve ter no máximo 20 caracteres")]
        public string Status { get; set; } = "PENDENTE"; // PENDENTE, REGISTRADO, LIQUIDADO, VENCIDO, CANCELADO

        // Campos de pagamento (persistidos da API Santander)
        /// <summary>
        /// Indica se o boleto foi realmente pago (LIQUIDADO ou BAIXADO com valor pago > 0)
        /// </summary>
        public bool FoiPago { get; set; } = false;

        /// <summary>
        /// Valor pago pelo cliente (retornado pela API Santander)
        /// </summary>
        public decimal? ValorPago { get; set; }

        /// <summary>
        /// Data em que o pagamento foi realizado
        /// </summary>
        public DateTime? DataPagamento { get; set; }

        // Mensagens do boleto (JSON)
        [StringLength(1000, ErrorMessage = "Mensagens devem ter no máximo 1000 caracteres")]
        public string? Messages { get; set; }

        // Número da parcela (ex: 1, 2, 3... para rastrear qual parcela do contrato)
        public int? NumeroParcela { get; set; }

        // ========================================================================
        // CAMPOS PARA BOLETO MANUAL - TIPO E PARCELAS VINCULADAS
        // ========================================================================
        
        /// <summary>
        /// Tipo de boleto manual: AVULSO, RENEGOCIACAO ou ANTECIPACAO
        /// NULL para boletos gerados automaticamente em lote
        /// </summary>
        [StringLength(20)]
        public string? TipoBoletoManual { get; set; }

        /// <summary>
        /// Tipo de pagamento: "Boleto" (padrão) ou "Pix"
        /// Quando "Pix", não é gerado boleto no Santander, apenas registro para controle de parcelas
        /// </summary>
        [StringLength(20, ErrorMessage = "Tipo de pagamento deve ter no máximo 20 caracteres")]
        public string TipoPagamento { get; set; } = "Boleto";
        
        /// <summary>
        /// Lista de parcelas cobertas por este boleto (JSON)
        /// Ex: "[5,6,7]" para renegociação das parcelas 5, 6 e 7
        /// </summary>
        [StringLength(500)]
        public string? ParcelasCobertas { get; set; }
        
        /// <summary>
        /// IDs dos boletos originais que foram renegociados (JSON)
        /// Ex: "[123,124,125]" para os boletos originais BAIXADO_NAO_PAGO
        /// </summary>
        [StringLength(500)]
        public string? BoletosOriginaisRenegociados { get; set; }

        // Campos de controle
        public DateTime DataCadastro { get; set; } = DateTime.Now;
        public DateTime? DataAtualizacao { get; set; }
        public bool Ativo { get; set; } = true;

        // Campos para rastreamento de erros
        [StringLength(10, ErrorMessage = "Código de erro deve ter no máximo 10 caracteres")]
        public string? ErrorCode { get; set; }

        [StringLength(500, ErrorMessage = "Mensagem de erro deve ter no máximo 500 caracteres")]
        public string? ErrorMessage { get; set; }

        [StringLength(50, ErrorMessage = "Trace ID deve ter no máximo 50 caracteres")]
        public string? TraceId { get; set; }

        // Campo para armazenar PDF no Azure Blob Storage (para portal do cliente)
        /// <summary>
        /// URL/caminho do PDF armazenado no Azure Blob Storage
        /// </summary>
        [StringLength(500, ErrorMessage = "URL do PDF deve ter no máximo 500 caracteres")]
        public string? PdfBlobUrl { get; set; }

        /// <summary>
        /// Data em que o PDF foi armazenado
        /// </summary>
        public DateTime? PdfArmazenadoEm { get; set; }
    }
}

File: frontend/src/types/boleto.ts
// Tipos para o módulo de boletos baseados no backend

export interface Boleto {
  id: number;
  contratoId: number;
  nsuCode: string;
  nsuDate: string;
  covenantCode: string;
  bankNumber: string;
  clientNumber?: string;
  dueDate: string;
  issueDate: string;
  nominalValue: number;
  documentKind: string;
  status: BoletoStatus;
  statusDescription?: string;

  // Campos de pagamento (novos)
  foiPago?: boolean;
  paidValue?: number;
  settlementDate?: string;
  paymentDate?: string;
  dataPagamento?: string; // Data de pagamento do banco (DateTime? formatado como string)

  // Dados do Pagador
  payerName: string;
  payerDocumentType: string;
  payerDocumentNumber: string;
  payerAddress: string;
  payerNeighborhood: string;
  payerCity: string;
  payerState: string;
  payerZipCode: string;

  // Dados de resposta da API Santander
  barCode?: string;
  digitableLine?: string;
  entryDate?: string;
  qrCodePix?: string;
  qrCodeUrl?: string;

  // Informações do Contrato
  contrato?: ContratoInfo;

  // Tipo do boleto manual (RENEGOCIACAO, ANTECIPACAO, AVULSO ou null para lote normal)
  tipoBoletoManual?: TipoBoletoManual | null;

  // Campos de controle
  dataCadastro: string;
  dataAtualizacao?: string;

  /** Indicador de vencimento conforme regra do backend (lista/detalhe). */
  estaVencido?: boolean;

  // Campos de erro
  errorCode?: string;
  errorMessage?: string;
  traceId?: string;
}

export interface ContratoInfo {
  id: number;
  numeroContrato: string;
  numeroPasta?: string;
  tipoServico?: string;
  clienteNome?: string;
  clienteDocumento?: string;
  valorContrato?: number;
  filialNome?: string;
  /** Quando a API anexa o contrato ao boleto (opcional). */
  situacao?: string;
}

// Tipos de protesto disponíveis
export type ProtestType =
  | "DIAS_CORRIDOS"
  | "DIAS_UTEIS"
  | "SEM_PROTESTO"
  | "NAO_PROTESTAR";

export interface CreateBoletoDTO {
  contratoId: number;
  dueDate: string;
  nominalValue: number;
  clientNumber?: string;

  // Campos opcionais para desconto, multa e juros
  finePercentage?: number;
  fineQuantityDays?: number;
  interestPercentage?: number;
  deductionValue?: number;
  writeOffQuantityDays?: number;

  // Campos de protesto (apenas para boletos manuais)
  protestType?: ProtestType;
  protestQuantityDays?: number;

  // Mensagens personalizadas
  messages?: string[];

  // Dados PIX (opcionais)
  pixKeyType?: PixKeyType;
  pixKey?: string;
  txId?: string;

  // Dados de desconto (opcionais)
  discount?: DescontoDTO;
}

export interface DescontoDTO {
  type: "VALOR_DATA_FIXA" | "PERCENTUAL_DATA_FIXA";
  discountOne?: DescontoValorDTO;
  discountTwo?: DescontoValorDTO;
  discountThree?: DescontoValorDTO;
}

export interface DescontoValorDTO {
  value: number;
  limitDate: string;
}

export type BoletoStatus =
  | "PENDENTE"
  | "REGISTRADO"
  | "ATIVO"
  | "LIQUIDADO"
  | "BAIXADO"
  | "VENCIDO"
  | "CANCELADO"
  | "ERRO";

export type PixKeyType =
  | "EMAIL"
  | "CPF"
  | "CNPJ"
  | "TELEFONE"
  | "CHAVE_ALEATORIA";

/** Resposta de GET /Boleto/dashboard (totais calculados no servidor). */
export interface DashboardFinanceiro {
  /** Aliases curtos retornados pela API (opcionais). */
  total?: number;
  valorTotal?: number;
  totalPago?: number;
  pendentes?: number;
  pagos?: number;
  vencidos?: number;

  totalBoletos: number;
  boletosPendentes: number;
  boletosRegistrados: number;
  boletosLiquidados: number;
  boletosVencidos: number;
  boletosCancelados: number;
  valorTotalRegistrado: number;
  valorTotalLiquidado: number;
  boletosHoje?: number;
  boletosEsteMes?: number;
}

export interface BoletoFilters {
  status?: BoletoStatus;
  contratoId?: number;
  clienteNome?: string;
  dataInicio?: string;       // Filtro por data de vencimento (início)
  dataFim?: string;          // Filtro por data de vencimento (fim)
  dataEmissaoInicio?: string; // Filtro por data de emissão (início)
  dataEmissaoFim?: string;    // Filtro por data de emissão (fim)
  valorMinimo?: number;
  valorMaximo?: number;
}

/** Totais da listagem calculados no backend (GET /Boleto, etc.). */
export interface BoletosListagemResumo {
  total: number;
  valorTotal: number;
  totalPago: number;
  pendentes: number;
  pagos: number;
  vencidos: number;
}

function pickNumericField(
  obj: Record<string, unknown>,
  keys: string[],
): number | null {
  for (const k of keys) {
    const v = obj[k];
    if (v === undefined || v === null || v === "") continue;
    const n = typeof v === "string" ? Number(v.replace(",", ".")) : Number(v);
    if (!Number.isNaN(n)) return n;
  }
  return null;
}

/** Converte o JSON do dashboard / resumo da API em {@link BoletosListagemResumo}. */
export function parseResumoPayload(raw: unknown): BoletosListagemResumo | null {
  if (!raw || typeof raw !== "object") return null;
  const r = raw as Record<string, unknown>;
  const total = pickNumericField(r, [
    "total",
    "Total",
    "totalBoletos",
    "TotalBoletos",
  ]);
  const valorTotal = pickNumericField(r, [
    "valorTotal",
    "ValorTotal",
    "valorTotalNominal",
    "ValorTotalNominal",
    "valorTotalRegistrado",
    "ValorTotalRegistrado",
  ]);
  const totalPago = pickNumericField(r, [
    "totalPago",
    "TotalPago",
    "valorTotalPago",
    "ValorTotalPago",
    "valorPago",
    "ValorPago",
    "valorTotalLiquidado",
    "ValorTotalLiquidado",
  ]);
  const pendentes = pickNumericField(r, [
    "pendentes",
    "Pendentes",
    "quantidadePendentes",
    "QuantidadePendentes",
    "boletosPendentes",
    "BoletosPendentes",
  ]);
  const pagos = pickNumericField(r, [
    "pagos",
    "Pagos",
    "quantidadePagos",
    "QuantidadePagos",
    "boletosPagos",
    "BoletosPagos",
    "boletosLiquidados",
    "BoletosLiquidados",
  ]);
  const vencidos = pickNumericField(r, [
    "vencidos",
    "Vencidos",
    "quantidadeVencidos",
    "QuantidadeVencidos",
    "boletosVencidos",
    "BoletosVencidos",
  ]);
  if (
    total === null &&
    valorTotal === null &&
    totalPago === null &&
    pendentes === null &&
    pagos === null &&
    vencidos === null
  ) {
    return null;
  }
  return {
    total: Math.round(total ?? 0),
    valorTotal: valorTotal ?? 0,
    totalPago: totalPago ?? 0,
    pendentes: Math.round(pendentes ?? 0),
    pagos: Math.round(pagos ?? 0),
    vencidos: Math.round(vencidos ?? 0),
  };
}

function unwrapListagemEnvelope(raw: unknown): unknown {
  let cur: unknown = raw;
  for (let depth = 0; depth < 4; depth++) {
    if (!cur || typeof cur !== "object" || Array.isArray(cur)) break;
    const o = cur as Record<string, unknown>;
    const inner = o.data ?? o.Data ?? o.result ?? o.Result;
    if (inner !== undefined && inner !== cur) {
      cur = inner;
      continue;
    }
    break;
  }
  return cur;
}

/**
 * Interpreta resposta da listagem de boletos: array legado ou envelope com lista + resumo.
 */
export function parseBoletosListResponse(raw: unknown): {
  boletos: Boleto[];
  resumo: BoletosListagemResumo | null;
} {
  const payload = unwrapListagemEnvelope(raw);
  if (Array.isArray(payload)) {
    return { boletos: payload as Boleto[], resumo: null };
  }
  if (!payload || typeof payload !== "object") {
    return { boletos: [], resumo: null };
  }
  const o = payload as Record<string, unknown>;
  const list =
    (Array.isArray(o.boletos) ? o.boletos : null) ??
    (Array.isArray(o.Boletos) ? o.Boletos : null) ??
    (Array.isArray(o.items) ? o.items : null) ??
    (Array.isArray(o.Items) ? o.Items : null) ??
    (Array.isArray(o.data) ? o.data : null) ??
    (Array.isArray(o.Data) ? o.Data : null);
  if (!list) {
    return { boletos: [], resumo: null };
  }
  const resumoRaw =
    o.resumo ?? o.Resumo ?? o.totais ?? o.Totais ?? o.summary ?? o.Summary;
  return {
    boletos: list as Boleto[],
    resumo: parseResumoPayload(resumoRaw),
  };
}

// Opções para status com cores e labels
export const BoletoStatusOptions = [
  {
    value: "PENDENTE" as BoletoStatus,
    label: "Pendente",
    color: "bg-gray-100 text-gray-800 border-gray-200",
  },
  {
    value: "REGISTRADO" as BoletoStatus,
    label: "Registrado",
    color: "bg-blue-100 text-blue-800 border-blue-200",
  },
  {
    value: "ATIVO" as BoletoStatus,
    label: "Ativo",
    color: "bg-blue-100 text-blue-800 border-blue-200",
  },
  {
    value: "LIQUIDADO" as BoletoStatus,
    label: "Pago (Boleto)",
    color: "bg-green-100 text-green-800 border-green-200",
  },
  {
    value: "BAIXADO" as BoletoStatus,
    label: "Baixado", // Pode ser pago via PIX ou expirado - depende de foiPago
    color: "bg-gray-100 text-gray-800 border-gray-200",
  },
  {
    value: "VENCIDO" as BoletoStatus,
    label: "Vencido",
    color: "bg-red-100 text-red-800 border-red-200",
  },
  {
    value: "CANCELADO" as BoletoStatus,
    label: "Cancelado",
    color: "bg-gray-300 text-gray-800 border-gray-400",
  },
  {
    value: "ERRO" as BoletoStatus,
    label: "Erro",
    color: "bg-pink-100 text-pink-800 border-pink-200",
  },
] as const;

/** Normaliza status vindo da API para comparação (evita "Liquidado" vs "LIQUIDADO"). */
export function normalizarStatusBoleto(
  status: Boleto["status"] | string | null | undefined
): string {
  return String(status ?? "")
    .toUpperCase()
    .trim();
}

/**
 * Normaliza `foiPago` da API. No banco/backend o campo é boolean (true/false);
 * `undefined` quando vier null/ausente → `verificarSeFoiPago` usa fallback por status (ex.: LIQUIDADO).
 */
export function parseFoiPagoApiValue(value: unknown): boolean | undefined {
  if (value === true) return true;
  if (value === false) return false;
  return undefined;
}

/**
 * Comissões: só contabiliza boleto com `foiPago === true`.
 * `false`, `null`, `undefined` ou ausente no JSON não entram (sem inferir por LIQUIDADO/BAIXADO).
 */
export function boletoPagoParaComissao(boleto: Pick<Boleto, "foiPago">): boolean {
  return boleto.foiPago === true;
}

/** Valor pago numérico a partir da API (aceita string "1500.00"). */
export function parsePaidValueApi(paid: unknown): number | undefined {
  if (paid === null || paid === undefined || paid === "") return undefined;
  const val =
    typeof paid === "string"
      ? Number(paid.replace(/\s/g, "").replace(",", "."))
      : Number(paid);
  if (Number.isNaN(val) || val <= 0) return undefined;
  return val;
}

// Função auxiliar para verificar se o boleto foi pago
// BAIXADO: só entra em totais/comissão com foiPago === true (ex.: PIX pago). Baixado com foiPago false ou ausente = não pago.
// LIQUIDADO: pago via código de barras costuma vir sem foiPago (fallback abaixo).
export function verificarSeFoiPago(boleto: Boleto): boolean {
  const st = normalizarStatusBoleto(boleto.status);

  if (st === "CANCELADO" || st === "ERRO") {
    return false;
  }

  if (st === "BAIXADO") {
    return boleto.foiPago === true;
  }

  if (boleto.foiPago === true) {
    return true;
  }

  if (boleto.foiPago === false) {
    return false;
  }

  if (st === "LIQUIDADO") {
    return true;
  }

  return false;
}

/**
 * Valor efetivo recebido no pagamento (comissões e totais devem usar isto quando existir).
 * Se a API não enviar paidValue ou for 0, usa nominalValue.
 */
export function valorEfetivoPagoBoleto(
  boleto: Pick<Boleto, "paidValue" | "nominalValue">
): number {
  const nominal = Number(boleto.nominalValue) || 0;
  const paid = parsePaidValueApi(
    (boleto as { paidValue?: unknown }).paidValue
  );
  if (paid !== undefined) return paid;
  return nominal;
}

// Função para obter a configuração de exibição do status
// IMPORTANTE: Usar foiPago como fonte da verdade
export function getStatusDisplayConfig(boleto: Boleto): {
  text: string;
  color: string;
  bgColor: string;
  isPago: boolean;
} {
  // 1. Primeiro verificar se foi pago usando o campo foiPago
  if (boleto.foiPago === true) {
    if (boleto.status === "LIQUIDADO") {
      return {
        text: "Pago (Cód. Barras)",
        color: "text-green-400",
        bgColor: "bg-green-500/20 border-green-500/30",
        isPago: true,
      };
    }
    if (boleto.status === "BAIXADO") {
      return {
        text: "Pago (PIX)",
        color: "text-green-400",
        bgColor: "bg-green-500/20 border-green-500/30",
        isPago: true,
      };
    }
    // Qualquer outro status com foiPago = true
    return {
      text: "Pago",
      color: "text-green-400",
      bgColor: "bg-green-500/20 border-green-500/30",
      isPago: true,
    };
  }

  // 2. Se não foi pago (foiPago === false ou undefined), verificar o status
  switch (boleto.status) {
    case "BAIXADO":
      // BAIXADO sem foiPago = Expirou após 30 dias sem pagamento
      return {
        text: "Baixado (Não Pago)",
        color: "text-orange-400",
        bgColor: "bg-orange-500/20 border-orange-500/30",
        isPago: false,
      };
    case "LIQUIDADO":
      // LIQUIDADO sem foiPago = Caso raro, mas mostrar como pago
      return {
        text: "Pago (Cód. Barras)",
        color: "text-green-400",
        bgColor: "bg-green-500/20 border-green-500/30",
        isPago: true,
      };
    case "ATIVO":
    case "REGISTRADO":
      return {
        text: "Aguardando",
        color: "text-blue-400",
        bgColor: "bg-blue-500/20 border-blue-500/30",
        isPago: false,
      };
    case "VENCIDO":
      return {
        text: "Vencido",
        color: "text-amber-400",
        bgColor: "bg-amber-500/20 border-amber-500/30",
        isPago: false,
      };
    case "CANCELADO":
      return {
        text: "Cancelado",
        color: "text-red-400",
        bgColor: "bg-red-500/20 border-red-500/30",
        isPago: false,
      };
    case "PENDENTE":
      return {
        text: "Pendente",
        color: "text-neutral-400",
        bgColor: "bg-neutral-700 border-neutral-600",
        isPago: false,
      };
    case "ERRO":
      return {
        text: "Erro",
        color: "text-pink-400",
        bgColor: "bg-pink-500/20 border-pink-500/30",
        isPago: false,
      };
    default:
      return {
        text: boleto.status || "Desconhecido",
        color: "text-neutral-400",
        bgColor: "bg-neutral-700 border-neutral-600",
        isPago: false,
      };
  }
}

export const PixKeyTypeOptions = [
  { value: "EMAIL" as PixKeyType, label: "E-mail" },
  { value: "CPF" as PixKeyType, label: "CPF" },
  { value: "CNPJ" as PixKeyType, label: "CNPJ" },
  { value: "TELEFONE" as PixKeyType, label: "Telefone" },
  { value: "CHAVE_ALEATORIA" as PixKeyType, label: "Chave Aleatória" },
] as const;

// ===== Tipos para Envio de Email =====

// Resposta do envio de email individual
export interface EmailEnvioInfo {
  enviado: boolean | null;
  destino: string | null;
  erro: string | null;
}

// Resposta da criação de boleto (POST /api/Boleto) com info de email
export interface CreateBoletoResponse {
  boleto: Boleto;
  email: EmailEnvioInfo | null;
}

// Resposta do endpoint de envio/reenvio de email
export interface EnviarEmailResponse {
  sucesso: boolean;
  erro: string | null;
  emailDestino: string | null;
}

// Request para enviar email com destino alternativo
export interface EnviarEmailRequest {
  emailDestino?: string;
}

// ===== Tipos para Boleto Manual (Renegociação/Antecipação/Avulso) =====

export type TipoBoletoManual = "AVULSO" | "RENEGOCIACAO" | "ANTECIPACAO";

export interface ParcelaDisponivelDTO {
  boletoId?: number;
  numeroParcela: number;
  valorOriginal: number;
  vencimentoOriginal: string;
  status: string;
  descricao: string;
}

export interface ParcelasDisponiveisResponse {
  parcelasRenegociacao: ParcelaDisponivelDTO[];
  parcelasAntecipacao: ParcelaDisponivelDTO[];
  valorParcela: number;
  totalParcelas: number;
  parcelaAtual: number;
  parcelasEmAtraso: number;
}

export interface ParcelaSelecionadaDTO {
  boletoId?: number;
  numeroParcela: number;
  valorOriginal: number;
  vencimentoOriginal?: string;
}

export interface CreateBoletoManualDTO extends CreateBoletoDTO {
  tipoBoletoManual?: TipoBoletoManual;
  parcelasSelecionadas?: ParcelaSelecionadaDTO[];
}
